enhancements/oc/cli-manager.md

```diff
@@ -0,0 +1,252 @@
+of services on OpenShift.  We need a mechanism for providing and consuming tools that is simple to add on to as new CLIs are
+developed from a variety of sources - and this should be specific for each cluster and available with disconnected installs.
+
+### Goals
+
+* No new form of binary distribution or binary creation will be proposed, because we have an existing structure at Red Hat.
+RPMs or images are the only options, and images must be deployed by the RH pipeline via operators. This proposal is for delivering
+CLIs via images, because this will enable offering CLIs offline through an existing image registry.
+* CLI owners must be able to easily distribute their binaries
+* The version of the CLI a user is offered is appropriate for the version of the CLI controller installed on the cluster
+* Arbitrary binaries not delivered by the in-cluster CRD controller are not important or relevant because of requirement 1 and 2
+  * Anyone who wants such tools can download them outside of this mechanism
+* CRD controller for registering CRs with an API for listing, extracting and downloading artifacts
+  * Krew-compatible index that both krew and oc can consume where the download links are for cluster-local resources
+  * The index and artifcats will served from a single route and service within the cluster
+
+### Non-Goals
+
+* `oc tools` will not build or serve the binaries.  It will know where to find them.
+* This proposal is not concerned with _which_ binaries will be managed.  This proposal is meant to determine the mechanism only.  Consumers and publishers are clients of the mechanism.
+* `oc tools` will not create or update the ClusterCLI Custom Resources, that will be managed by individual CLIs. However, the CLIs that are
+currently available as `consoleclidownloads`, `oc, odo, and helm`, will be included in the reference implementation of this enhancement.
+
+## Proposal
+
+Each component wishing to provide customers with their binaries will build and publish artifacts via an official channel to a central index.
+Each component wishing to provide customers with binaries will create a ClusterCLI custom resource to provide an image name and the file path
+within that image for the binaries. The CLIs currently offered via console-cli-downloads will be included in the reference implementation of this enhancement.
+`oc` will gather ClusterCLI from the cluster, and also the fileserver will offer the binaries via the console by mounting the binaries
+from the location/image in each custom resource.
+
+CLIs could be installed in a similar manner to plugins through [krew](https://github.com/kubernetes-sigs/krew).
+CLI images would serve the artifacts to enable disconnected downloads, and OpenShift CLIs will be available as
+[krew plugins](https://github.com/kubernetes-sigs/krew-index/tree/master/plugins).
+
+* Krew and Krew plugins are upstream projects that Kubernetes users are already familiar with
```

Comment:
I'd suggest the proposal section be shaped in the following way:
1. introduce what krew is and how it fits the image of delivering plugins based on the index metadata, and most importantly why we picked krew instead of creating our own mechanism 
2. describe how we'll serve the index metadata for 1 inside the cluster
3. describe how a binary author is meant to have his binary served by 2. 

First describe the process from a high level, then create a detailed section for each of them drilling down into implementation details.